app/controllers/application_controller.rb

```diff
@@ -5,6 +5,146 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "unbreakable_password"
   end
 
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if logged_in? 
+      redirect '/tweets'
+    else
+      erb :'/users/create_user'
+    end
+  end
+
+  post '/signup' do
+    user = User.new(params) 
+    if !user.attributes.values.include?("")&& user.save
+      session[:user_id] = user.id
+      redirect '/tweets'
+    else
+      redirect '/signup'
+    end
+  end
+
+  get '/login' do
+    if logged_in? 
+      redirect '/tweets'
+    else
+      erb :'/users/login'
+    end  
+  end
+
+  post '/login' do
```

Comment:
you should set up a verification gate here to make sure a User is not logged_in? before they create another account. Otherwise a user could make a user by changing the parameters in a request.